### PR TITLE
Run full functional tests for CONTAINER_IMAGE component

### PR DESCRIPTION
**Pull request by sshnaidm**
Run functional tests for Podman related code in CI.

### CODE REVIEW

File: .github/workflows/pull-request-podman.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          sudo journalctl -u directord-server -n 2000 &> /tmp/directord-server.log || true
+          sudo /opt/directord/bin/directord manage --list-nodes &> /tmp/directord-nodes.log
+          sudo /opt/directord/bin/directord manage --dump-cache &> /tmp/directord-cache.log
+          sudo /opt/directord/bin/directord manage --export-jobs /tmp/directord-jobs-export.log
+        if: failure()
+      - name: Upload build Log artifacts on failure
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: functional-realworld-test-logs
```
Comment (cloudnull) [inline .github/workflows/pull-request-podman.yml:68]: The name of this should match the job name `functional-podman-functional`.

File: .github/workflows/pull-request-podman.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        run: |
+          sudo /opt/directord/bin/directord-server-systemd
+          sudo systemctl daemon-reload
+          sudo systemctl restart directord-server
+      - name: Build current container and run client
+        run: |
+          sudo mkdir /tmp/containers
+          sudo apt-get install -y podman
+          sudo podman build -t test-image .
+          sudo podman run --hostname container-client \
```
Comment (cloudnull) [inline .github/workflows/pull-request-podman.yml:33]: do we need to run a directord client container here? Feels like we can just run podman directly like so: https://github.com/directord/directord/blob/main/.github/workflows/pull-request.yml#L548-L577

**Review by cloudnull — CHANGES_REQUESTED** [review]

**Comment by cloudnull** [comment]
Also we should get a commit message, so that this is all captured in our release note generation.

File: .github/workflows/pull-request-podman.yml (hunk 1 above)
Comment (sshnaidm) [reply .github/workflows/pull-request-podman.yml:68]: ack

File: .github/workflows/pull-request-podman.yml (hunk 2 above)
Comment (sshnaidm) [reply .github/workflows/pull-request-podman.yml:33]: I can change it, but this is more fun! :rofl:

**Pull request merged**